exporter/awsemfexporter/metric_translator.go

```diff
@@ -118,6 +136,253 @@ func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
 	return dps.DoubleHistogramDataPointSlice.At(i)
 }
 
+
+// getNamespace retrieve namespace for given set of metrics from user config 
+func getNamespace(rm *pdata.ResourceMetrics, namespace string) (string) {
+	if len(namespace) == 0 {
+		serviceName, svcNameOk := rm.Resource().Attributes().Get(conventions.AttributeServiceName)
+		serviceNamespace, svcNsOk := rm.Resource().Attributes().Get(conventions.AttributeServiceNamespace)
+		if svcNameOk && svcNsOk && serviceName.Type() == pdata.AttributeValueSTRING && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = fmt.Sprintf("%s/%s", serviceNamespace.StringVal(), serviceName.StringVal())
+		} else if svcNameOk && serviceName.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceName.StringVal()
+		} else if svcNsOk && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceNamespace.StringVal()
+		}
+	}
+
+	if len(namespace) == 0 {
+		namespace = defaultNameSpace
+	}
+	return namespace
+}
+
+// TranslateOtToGroupedMetric converts OT metrics to GroupedMetric format
+func TranslateOtToGroupedMetric(metric pdata.Metrics, config *Config) (map[string]*GroupedMetric, int) {
+	totalDroppedMetrics := 0
+	groupedMetricMap := make(map[string]*GroupedMetric)
+	timestamp := time.Now().UnixNano() / int64(time.Millisecond)
```

Comment:
I'm thinking we should use timestamp [here](https://github.com/open-telemetry/opentelemetry-proto/blob/master/opentelemetry/proto/metrics/v1/metrics.proto#L326) (if exists) prior to use `now`

Reply:
yes